docs-examples/inject_file/src/main.rs

```diff
@@ -0,0 +1,68 @@
+use std::{borrow::Cow, fs::read_to_string};
+
+use regex::{Captures, Regex};
+
+fn main() {
+    let args = std::env::args().collect::<Vec<_>>();
+    let input_path = args
+        .get(1)
+        .expect("Expected 1 argument: the path to the file to parse.");
+    let file: &str = &read_to_string(&input_path)
+        .expect(&format!("Could not read file at input {}", input_path));
+
+    let result = injected(file, |file_path| {
+        let mut path = "..".to_string();
+        path.push_str(&file_path);
+        path.to_string()
+    });
+    print!("{result}");
+}
+
+fn injected(source_text: &str, get_path: fn(&str) -> String) -> Cow<str> {
+    // Regex to find "<load" tags and capture their info (path + marker)
+    let re = Regex::new(r"<load path='(.*)'.*marker='(.*)'.*>").unwrap();
+
+    let result = re.replace_all(source_text, |caps: &Captures| {
+        let infos = &caps.extract::<2>().1;
+        assert!(
+            infos.len() == 2,
+            "load tag encountered without a path or marker"
+        );
+        let path = get_path(infos[0]);
+        // Reading file from the path of the tag of input file
+        let to_inject = read_to_string(&path).expect(&format!("could not read path: {}", path));
+
+        // Regex to find the markers inside comments, and only print what's inside
+        // FIXME: I think we should just paste all the inside,
+        // and then remove all "// DOCUSAURUS*"" lines, to allow reuse of a same file.
```

Comment:
removing the `// DOCUSAURUS` is quite important. Or rework the early portions I did (rigidbody) where one similar piece of code could be used in multiple places.

Reply:
done